api/src/apps/apps.rs

```diff
@@ -285,6 +283,46 @@ impl AppsService {
         Ok(services)
     }
 
+    /// Adds all companions to service_configs that are not yet present. Also updates
+    /// service configs if there is a corresponding companion config.
+    fn merge_companions_by_service_name(
+        companions: Vec<ServiceConfig>,
+        service_configs: &mut Vec<ServiceConfig>,
+    ) -> () {
+        for service in service_configs.iter_mut() {
+            let matching_companion = companions
+                .iter()
+                .find(|companion| companion.service_name() == service.service_name());
+
+            if matching_companion.is_some() {
+                AppsService::merge_service_configs(service, matching_companion.unwrap());
+            }
+        }
+
+        service_configs.extend(AppsService::filter_companions_by_service_name(
+            companions,
+            service_configs,
+        ));
+    }
+
+    fn merge_service_configs(to: &mut ServiceConfig, from: &ServiceConfig) -> () {
+        // We append the to-envvars to the from-envvars. If the same value
+        // is set twice, docker takes the most recent one.
+        // This way we let docker handle parsing the variables which means
+        // less room for errors.
+        let mut envs = from.env().unwrap_or(&vec![]).clone();
```

Comment:
I would prefer not to rely on Docker here because another infrastructure could reject the deployment. Therefore, the code should prefer the environment variables of the request payload.

Reply:
The environment variables are currently given as strings of the form `VAR=123` without parsing what the variable and what the allocated value is. As far as I understand this string is given unmodified to Docker without parsing or modifying it. As a consequence, there already is an implicit reliance on Docker with regards to the format of the environment variable string. Some other infrastructure might for example need a format like `VAR '123'` which would be possible with the current implementation, as the string is passe on to the infrastructure without modification.

If we wanted to parse the string however, we would need to define the format supported by PREvant explicitly.

Reply:
That's true. However, I'm already working on it to get rid of this assumption in the Kubernetes branch. In this branch PREvant parses `XXX=YYY` into key and value and it should be merged in the near future. 

Until then you could make here an simple assumption and split by `=` and search for overlapping keys.

Reply:
Alright, I added parsing of environment variables. To avoid breaking compatibility with possible other formats, I added a fallback if an env-string is not of the form `A=B`. I also added tests for the `merge_with` function in `ServiceConfig`. Let me know if you have any more feedback!

Reply:
Alright, I added parsing of environment variables. To avoid breaking compatibility with possible other formats, I added a fallback if an env-string is not of the form `A=B`. I also added tests for the `merge_with` function in `ServiceConfig`. Let me know if you have any more feedback!